cmd/kubectl-gadget/trace/tcpdump.go

```diff
@@ -0,0 +1,399 @@
+// Copyright 2019-2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	"github.com/google/gopacket/pcapgo"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	commontrace "github.com/inspektor-gadget/inspektor-gadget/cmd/common/trace"
+	commonutils "github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/kubectl-gadget/utils"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/tcpdump/types"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/k8sutil"
+	eventtypes "github.com/inspektor-gadget/inspektor-gadget/pkg/types"
+)
+
+type Decoder string
+
+type nameEvent struct {
+	IP   net.IP
+	Name string
+}
+
+type TCPDumpParser struct {
+	commonutils.BaseParser[types.Event]
+	pcapngWriter   *pcapgo.NgWriter
+	writer         io.Writer
+	decoder        Decoder
+	snapLen        int
+	filter         string
+	interfaces     map[string]int
+	interfacesLock sync.RWMutex
+	events         chan *types.Event // channel for incoming packets
+	nameEvents     chan *nameEvent   // channel for dns resolution information
+}
+
+const (
+	DecoderWireshark = "wireshark"
+	DecoderTCPDump   = "tcpdump"
```

Comment:
While tracing with `./kubectl-gadget trace tcpdump --decoder localhost` I am seeing `localhost` everywhere:
```
→ ./kubectl-gadget trace tcpdump --decoder tcpdump
reading from file -, link-type EN10MB (Ethernet)
17:29:20.556811 IP localhost.60785 > localhost.domain: 24363+[|domain]
17:29:20.556913 IP localhost.60785 > localhost.domain: 25724+[|domain]
17:29:20.557411 IP localhost.domain > localhost.60785: 24363*-[|domain]
17:29:20.557566 IP localhost.domain > localhost.60785: 25724*-[|domain]

```

Reply:
I also got a lot of localhost packets - but those seem valid (from 127.0.0.1 to 127.0.0.1). Could you please recheck to see if you also see inbound/outbound packets when doing a curl or such?

The interface name would greatly help to show from which container those local packets are coming, but that's only visible in WireShark as it seems.

Reply:
I am running ` nslookup kubernetes.default.svc.cluster.local` in a test pod with `alpine` image. If try to see the IP addresses I am seeing following:
```
→ ./kubectl-gadget trace tcpdump --decoder tcpdump --decoder-args -n 
reading from file -, link-type EN10MB (Ethernet)
16:49:59.081782 IP 172.17.0.2.33583 > 10.96.0.10.53: 60708+[|domain]
16:49:59.081894 IP 172.17.0.2.33583 > 10.96.0.10.53: 61948+[|domain]
16:49:59.100071 IP 10.96.0.10.53 > 172.17.0.2.33583: 60708 1/0/0 (54)
16:49:59.100731 IP 10.96.0.10.53 > 172.17.0.2.33583: 61948 1/0/0 (66)
```
So I do see a DNS requests going towards `kube-dns` service but not sure why they are named localhost. I will check again if the setup has some weird config. It is minikube node.